spec/comment_spec.rb

```diff
@@ -83,5 +93,27 @@
         @comments.should_not include(@other_comment)
       end
     end
+    
+    describe "#visible" do
+      before :each do
+        @hidden_comment = Comment.create!(:body => 'hidden', :user => @user)
+        @child_comment = Comment.create!(:body => 'child of hidden', :user => @user, :parent => @hidden_comment)
+        @hidden_comment.update_attributes(:hidden => true)
+
+        @comments = Comment.visible
+      end
+
+      it "should return the comments that have not been flagged as hidden" do
+        @comments.should include(@comment)
+      end
+
+      it "should not return the comments that have been flagged as hidden" do
+        @comments.should_not include(@hidden_comment)
+      end
+      
+      it "should not return comments that are children of comments that have been hidden" do
```

Comment:
```suggestion
      it 'does return the comments that have not been flagged as hidden' do
        @comments.should include(@comment)
      end

      it 'does not return the comments that have been flagged as hidden' do
        @comments.should_not include(@hidden_comment)
      end
      
      it 'does not return comments that are children of comments that have been hidden' do
```
Do not use 'should': https://www.betterspecs.org/#should